src/CMakeLists.txt

```diff
@@ -74,10 +74,16 @@ if (ENABLE_PROFILER)
     target_link_libraries(decomp2d PRIVATE caliper)
   endif()
 endif()
+
 if (HALO_DEBUG)
   target_compile_definitions(decomp2d PUBLIC -DHALO_DEBUG)
 endif()
 
+if (IO_BACKEND MATCHES "adios2")
+  target_compile_definitions(decomp2d PUBLIC -DADIOS2)
+  target_link_libraries(decomp2d PRIVATE adios2::fortran_mpi adios2::fortran)
```

Comment:
could/should this be `PUBLIC`?

Reply:
Yes the PUBLIC seems to be the better solution 
No idea why with gnu is OK